functests/common.go

```diff
@@ -4,28 +4,252 @@ import (
 	"fmt"
 	"os"
 
+	"k8s.io/klog"
+	"k8s.io/kubectl/pkg/scheme"
+
+	cincinnativ1beta1 "github.com/openshift/cincinnati-operator/pkg/apis/cincinnati/v1beta1"
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 )
 
-const operatorName = "openshift-cincinnati"
-
-// GetK8sClient is the function used to retrieve the kubernetes client
-func GetK8sClient() (*kubernetes.Clientset, error) {
+// GetConfig is the function used to retrieve the kubernetes config
+func GetConfig() (*rest.Config, error) {
 	kubeconfig := os.Getenv("KUBECONFIG")
 	if kubeconfig == "" {
 		return nil, fmt.Errorf("No KUBECONFIG environment variable set")
 	}
-
 	// K8s Core api client
 	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
 	if err != nil {
 		return nil, err
 	}
-	k8sClient, err := kubernetes.NewForConfig(config)
+	return config, nil
+}
+
+// GetK8sClient is the function used to retrieve the kubernetes client
+func GetK8sClient() (*kubernetes.Clientset, error) {
+	config, err := GetConfig()
```

Comment:
When I don't have KUBECONFIG set in my environment. I see this error: 
````
$ make func-test
Running functional test suite
hack/functest.sh
GINKO binary found at /home/rwsu/go/bin/ginkgo
Compiling functests...
    compiled functests.test
panic: 
Your test failed.
Ginkgo panics to prevent subsequent assertions from running.
Normally Ginkgo rescues this panic so you shouldn't see it.

But, if you make an assertion in a goroutine, Ginkgo can't capture the panic.
To circumvent this, you should call

	defer GinkgoRecover()

at the top of the goroutine that caused this panic.


goroutine 1 [running]:
github.com/onsi/ginkgo.Fail(0xc0003ae580, 0x3d, 0xc00070eb40, 0x1, 0x1)
	/home/rwsu/go/pkg/mod/github.com/onsi/ginkgo@v1.12.0/ginkgo_dsl.go:267 +0xc8
github.com/onsi/gomega/internal/assertion.(*Assertion).match(0xc000729300, 0x170f5c0, 0x220a2b0, 0x0, 0x0, 0x0, 0x0, 0xc000729300)
	/home/rwsu/go/pkg/mod/github.com/onsi/gomega@v1.10.0/internal/assertion/assertion.go:79 +0x20e
github.com/onsi/gomega/internal/assertion.(*Assertion).NotTo(0xc000729300, 0x170f5c0, 0x220a2b0, 0x0, 0x0, 0x0, 0xc00009a060)
	/home/rwsu/go/pkg/mod/github.com/onsi/gomega@v1.10.0/internal/assertion/assertion.go:48 +0xc7
github.com/openshift/cincinnati-operator/functests.glob..func5()
	/home/rwsu/go/src/github.com/openshift/cincinnati-operator/functests/podDisruptionBudget_test.go:17 +0x131
github.com/onsi/ginkgo/internal/suite.(*Suite).PushContainerNode(0xc000146fa0, 0x152b9c6, 0x13, 0x15be4c0, 0x0, 0x20a51c7, 0x60, 0xd, 0xc00002c370, 0xa5)
	/home/rwsu/go/pkg/mod/github.com/onsi/ginkgo@v1.12.0/internal/suite/suite.go:144 +0x158
github.com/onsi/ginkgo.Describe(0x152b9c6, 0x13, 0x15be4c0, 0x0)
	/home/rwsu/go/pkg/mod/github.com/onsi/ginkgo@v1.12.0/ginkgo_dsl.go:294 +0xaa
github.com/openshift/cincinnati-operator/functests.init()
	/home/rwsu/go/src/github.com/openshift/cincinnati-operator/functests/podDisruptionBudget_test.go:13 +0x1d7

Ginkgo ran 1 suite in 3.214910501s
Test Suite Failed
make: *** [Makefile:12: func-test] Error 1
````

Reply:
Done